lms/djangoapps/verified_track_content/tasks.py

```diff
@@ -30,31 +31,43 @@ def sync_cohort_with_mode(self, course_id, user_id, verified_cohort_name, defaul
         enrollment = CourseEnrollment.get_enrollment(user, course_key)
         # Note that this will enroll the user in the default cohort on initial enrollment.
         # That's good because it will force creation of the default cohort if necessary.
-        current_cohort = get_cohort(user, course_key)
+        try:
+            current_cohort = get_cohort(user, course_key)
+        except IntegrityError as integrity_error:
+            # It is quite common that get_cohort will throw an IntegrityError. This happens
+            # when 2 celery workers are both handling enrollment change events for the same user
+            # (for example, if the enrollment mode goes from None -> Audit -> Honor); if the user
+            # was not previously in a cohort, calling get_cohort will result in a cohort assignment.
+            LOGGER.info(
+                "HANDLING_INTEGRITY_ERROR: IntegrityError encountered for course '%s' and user '%s': %s",
+                course_id, user.id, unicode(integrity_error)
+            )
+            current_cohort = get_cohort(user, course_key)
```

Comment:
As we discussed, the sleep should not be necessary. When the IntegrityError is hit, it is because the "same" entry already exists in the database, so fetching it again will solve the issue.

I can verify this on stage though once it is up (since sandboxes have a single celery worker).